### PR TITLE
Fix some CMake target link libraries

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

The recent test refactor changed some libraries around, and did not always correctly add/update dependencies. On my system, this causes builds to emit abundant warnings from boost and yaml includes.

This PR fixes (some) `target_link_libraries` to address warnings on my system. Presumably, there are other places where the dependencies need fixing as well, but this PR does not claim to fix all dependencies.

### Types of changes:

- [x] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [x] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by fmahebert** [comment]
@carmaza FYI, these changes fixed the warnings I saw. Perhaps they will help in your case too?

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Thank you for doing this :)

**Comment by carmaza** [comment]
After rebasing on develop, upgrading to Blaze 3.7 (I use Spack), cherry-picking this branch, and building in my local environment using both clang and gcc, I no longer see any warnings. Thanks so much!

**Review by carmaza — APPROVED** [review]

**Comment by fmahebert** [comment]
@carmaza Great, thanks for checking!

@nilsdeppe Thanks for helping me figure out what was going on here 🙂

**Review by kidder — APPROVED** [review]

File: tests/Unit/ParallelAlgorithms/EventsAndTriggers/CMakeLists.txt [hunk 1]
```diff
@@ -19,3 +19,9 @@ add_dependencies(
   ${LIBRARY}
   module_ConstGlobalCache
   )
+
```
Comment (kidder) [inline tests/Unit/ParallelAlgorithms/EventsAndTriggers/CMakeLists.txt:22]: so let's get this merged to fix the problem, but going forward is the plan to eliminate `add_test_library` as I don't know if I am supposed to add libs to the final argument of that or to `target_link_libraries`...

Comment (nilsdeppe) [reply tests/Unit/ParallelAlgorithms/EventsAndTriggers/CMakeLists.txt:22]: The plan at least is to no longer specify the  `target_link_libraries` as part of the call, they should be added after :)

**Pull request merged**